src/main/java/org/axonframework/intellij/ide/plugin/publisher/DefaultEventPublisher.java

```diff
@@ -20,8 +20,9 @@ public boolean canPublishType(PsiType eventType) {
                 && publishedType != null
                 && eventType.isValid()
                 && (eventType.isAssignableFrom(publishedType)
-                || eventType instanceof PsiImmediateClassType
-                && ((PsiImmediateClassType) eventType).getParameters()[0].isAssignableFrom(publishedType));
+                || (eventType instanceof PsiImmediateClassType
```

Comment:
I'm not really sure about the intention of this code to perform a meaningful refactor... The line I added "((PsiImmediateClassType) eventType).getParameterCount() > 0" - was based on getting an IndexOutOfBoundsException from the line 24